src/js/jobs/utils.js

```diff
@@ -1,233 +1,47 @@
+export const getStepDescription = step => {
+    if (step.step_name && step.step_description) {
+        return {
+            title: step.step_name,
+            description: step.step_description
+        };
     }
-};
-
-const nuvsDescriptions = {
-    make_analysis_dir: makeAnalysisDirDescription,
-    prepare_reads: prepareReadsDescription,
-    prepare_qc: prepareQCDescription,
-    eliminate_otus: {
-        title: "Eliminate OTUs",
-        description: "Map against the OTU index and eliminate reads that match."
-    },
-    eliminate_subtraction: {
-        title: "Eliminate Subtraction",
-        description: "Map against the subtraction index and eliminate reads that match."
-    },
-    reunite_pairs: {
-        title: "Reunite Pairs",
-        description: "Ensure all remaining read pairs are paired correctly after subtraction."
-    },
-    assemble: {
-        title: "Assemble",
-        description: "Assemble contigs from remaining reads using SPAdes."
-    },
-    process_fasta: {
-        title: "Process Assembly",
-        description: "Find significant open reading frames (ORFs) in the assembled contigs."
-    },
-    prepare_hmm: {
-        title: "Prepare HMM Profiles",
-        description: "Run hmmpress on profiles in preparation for running hmmscan."
-    },
-    vfam: {
-        title: "VFam",
-        description: "Use hmmscan and vfam profiles to find viral motifs in the assembled contigs."
-    },
-    import_results: importResultsDescription
-};
-
-export const stepDescriptions = {
-    aodp: aodpDescriptions,
-    build_index: buildIndexDescriptions,
-    create_sample: createSampleDescriptions,
-    create_subtraction: createSubtractionDescriptions,
-    pathoscope_bowtie: pathoscopeBowtieDescriptions,
-    nuvs: nuvsDescriptions
-};
-
-export const getStepDescription = (stage, state, workflow) => {
-    if (state === "waiting") {
+    if (step.state === "waiting") {
         return {
             title: "Waiting",
             description: "Waiting for resources to become available."
         };
     }
 
-    if (state === "preparing") {
+    if (step.state === "preparing") {
         return {
             title: "Preparing",
             description: "Preparing the workflow environment."
         };
     }
 
-    if (state === "complete") {
+    if (step.state === "complete") {
         return {
             title: "Complete",
             description: ""
         };
     }
 
-    if (state === "cancelled") {
+    if (step.state === "cancelled") {
         return {
             title: "Cancelled",
             description: ""
         };
     }
 
-    if (state === "terminated") {
+    if (step.state === "terminated") {
         return {
             title: "Terminated",
             description: "There was a system malfunction"
         };
     }
 
-    return get(stepDescriptions, [workflow, stage], {
-        description: "",
-        title: startCase(stage)
-    });
+    return {
+        title: "",
```

Comment:
I think it is still a good idea return the stage as the title as a fallback if none of the other conditions are met.
```suggestion
        title: startCase(stage),
```

Reply:
The API can be relied on to provide a `step_name`. Keep the empty string so it is obvious if this breaks.